src/python/zquantum/core/circuit/_gate.py

```diff
@@ -888,3 +915,116 @@ def _unitary_(self):
+    An Orquestra gate class that deals with synthesis of a Multi Toffoli Gate
+    Although this inherits the Gate class it has extra attributes fundamentally related to
+    the fact that the Multi Toffoli Gate is a circuit rather a gate applied natively on a device
+
+    Attributes:
+    circuit (Orquestra Circuit): Needed to apply the mct gate from qiskit
+
+    control_qubits (list): This list contains integer labels for the qubits thats will act as the
+    control qubits. e.g [1, 2, 3, 4]
+    target_qubits (list): This is a list that contains integer labels for the qubits that are target qubits
+
+    all_qubits (list): A list of zmachine.core.qubit.Qubit This is necessary to determine what qubits
+    are the ancilla
+
+    ancilla_qubits (list): A list that contians integer labels for all the qubits labels that are the ancilla. These are needed since the MCT synthesis
+    requires in general addition of ancilla qubits
+
+    ccx_list (list): A list that contains CCX orquestra gates that represent the decompostion of the MCT gate
+
+    ccx_decomposition (qiskit circuit): Contains the Toffoli Decomposition which is the output from the mct gate
+
+    qiskit_ctrl_q (list): List of qiskit qubits that are control qubits for the MCT gate
+    qiskit_targ_q (list): List of qiskit qubits that are targ qubits for the MCT gate
+
+    qiskit_ancilla_q (list): List of qiskit qubits that are ancilla qubits for the mct gate synthesis
+
+
+    """
+
+    def __init__(self, circuit, control_qubits=[], target_qubits=[]):
+        if isinstance(control_qubits, list) and isinstance(target_qubits, list):
+            self.control_qubits = control_qubits
+            self.target_qubits = target_qubits
+            self.circuit = circuit
+            self.all_qubits = self.circuit.qubits
+            self.ancilla_qubits = []
+            self.ccx_list = []
+            self.qiskit_data = []
+            for i in range(len(self.all_qubits)):
+                if i not in self.control_qubits and i not in self.target_qubits:
+                    self.ancilla_qubits.append(i)
+        else:
+            raise TypeError('Expecting a list integer labels for control and target qubits')
+        # This is the ibm circuit object with all the toffoli gates
+        self.ccx_decomposition = qiskit.QuantumCircuit(len(self.all_qubits), len(self.all_qubits))
+        self.qiskit_ctrl_q = None
+        self.qiskit_targ_q = None
+        self.qiskit_ancilla_q = None
+
+    # super().__init__('MCT', qubits=self.all_qubits, control_qubits=control_qubits, target_qubits=target_qubits)
+    def _synthesize_circuit(self):
+        """
+         Produces the gate decomposition using qiskit
+        """
+        self.ccx_decomposition.mct(self.qiskit_ctrl_q, self.qiskit_targ_q, self.qiskit_ancilla_q)
+    def _create_qiskit_qubits(self):
+        """
+        Producing list of qiskit qubit objects
+        """
+        qreg = QuantumRegister(len(self.circuit.qubits), 'q')
```

Comment:
It seems to me that:
a) You could have used `self.all_qubits` here instead of `self.circuit.qubits`
b) You need to have `circuit` only to get the qubits – so actually, you don't need to pass `circuit` as a parameter (which I really don't like), you just need to pass `qubits`, right?